src/commands/signzone.rs

```diff
@@ -0,0 +1,4157 @@
+            // private key which tests that they match.
+            for public_key in &found_public_keys {
+                // Attempt to create a key pair from this public key and every
+                // private key that we have.
+                if let Ok(signing_key) = self.mk_signing_key(
+                    (*public_key.owner()).clone(),
+                    &private_key,
+                    (*public_key.data()).clone(),
+                ) {
+                    // Match found, keep the created signing key.
+                    // TODO: Log here.
+                    // TODO: Check the key tag against the key tag in the key file name?
+                    // println!(
+                    //     "DNSKEY RR with key tag {} matches loaded private key '{}'",
+                    //     public_key.key_tag(),
+                    //     private_key_path.display()
+                    // );
+                    signing_keys.push(signing_key);
+                    continue 'next_key_path;
+                }
+            }
+
+            // No matching public key found, try to load the public key
+            // instead.
+            let public_key_path = Self::mk_public_key_path(&key_path);
+            let public_key = Self::load_public_key(&env.in_cwd(&public_key_path))?;
+
+            // Verify that the owner of the public key matches the apex of the
+            // zone.
+            if !self.no_require_keys_match_apex && public_key.owner() != soa_rr.owner() {
+                return Err(format!(
+                    "Zone apex ({}) does not match the expected apex ({})",
+                    soa_rr.owner(),
+                    public_key.owner()
+                )
+                .into());
+            }
+
+            // Attempt to crate a key pair from the loaded private and public
+            // keys.
+            let signing_key = self
+                .mk_signing_key(
+                    public_key.owner().clone(),
+                    &private_key,
+                    public_key.data().clone(),
+                )
+                .map_err(|err| {
+                    format!(
+                        "Unable to create key pair from '{}' and '{}': {}",
+                        public_key_path.display(),
+                        private_key_path.display(),
+                        err
+                    )
+                })?;
+
+            // Store the created signing key.
+            signing_keys.push(signing_key);
+
+            // TODO: Log
+            // println!(
+            //     "Loaded public key with key tag {} from '{}' for private key '{}'",
+            //     public_key.key_tag(),
+            //     public_key_path.display(),
+            //     private_key_path.display()
+            // );
+        }
+
+        // First split the key into Key Signing Keys (KSK) that sign the
+        // DNSKEY RRset and Zone Signing Keys (ZSK) that sign the zone.
```

Comment:
This comment is outdated.
```suggestion
        // First split the key into Key Signing Keys (KSK) that sign apex
        // records involved in the chain of trust, and Zone Signing Keys (ZSK)
        // that sign the rest of the zone.
```

Reply:
I think this is wrong (or at least confusing) for CSKs